### ISSUE TITLE
[BUG] @NonNull not applied to "plural" builder method if marked with @Singular

### ISSUE THREAD

**Issue by stevenpaligo** [rzwitserloot/lombok#1999]
**Describe the bug**
When a builder method is created for a collection property that is marked @ Singular and @ NonNull, the generated "singular" method has a null check, but the "plural" method doesn't.  

**To Reproduce**
```
import java.util.Set;
import lombok.Builder;
import lombok.NonNull;
import lombok.Singular;

public class BugDemo {

  @Builder
  public BugDemo(@NonNull @Singular Set<@NonNull String> values) {
    System.out.println("Hello world!");
  }
}
```

**Expected behavior**
I would expect both methods to have the null check. When I decompile the code, only the "singular" method does:
```
public BugDemoBuilder values(Collection<? extends String> values)
{
  if (this.values == null) {
    this.values = new ArrayList();
  }
  this.values.addAll(values);return this;
}

public BugDemoBuilder value(@NonNull String value)
{
  if (value == null) {
    throw new IllegalArgumentException("value is marked @NonNull but is null");
  }
  if (this.values == null) {
    this.values = new ArrayList();
  }
  this.values.add(value);return this;
}
```

**Version info (please complete the following information):**
 - Lombok version: 1.18.4
 - Platform: javac 1.8.0_162

**Additional context**
I have Lombok configured to use IllegalArgumentException instead of NullPointerException, if that matters.

**Comment by rspilker** [comment]
Actually, I would expect:

```java
public BugDemoBuilder values(Collection<? extends @NonNull String> values)
{
  if (this.values == null) {
    this.values = new ArrayList();
  }
  this.values.addAll(values);return this;
}

```

**Comment by rspilker** [comment]
Ah, well, you cannot "set" the plural, only add the elements from the collection provided.

So this is not a bug, it is a feature. Calling it with `null` means, don't add anything. There will be no exception.

The constructor will never be called with a null list though.

**Comment by rspilker** [comment]
Okay, the  latest version did add the `@NonNull` to `Collection<? extends @NonNull String>`.

But I also see that we:

- Don't add @NonNull to the builder
- But throw an NPE if a null collection is passed, because we pass it to addAll

That's not good.

**Comment by rspilker** [comment]
The problem is that the parameter of the generated method isn't semantically the same as the `@Singular` annotated parameter. So we cannot just copy the `@NonNull`.

Generating an `@NonNull` for documentation purposes is a bit strange, especially since we now support a wider range of non-null types. Which one should we generate? Wecurrently don't have a system for that.

We could improve the error message a bit by generating a better exception, instead of leaving it up to the `addAll`.

I did have a look at the code, and although doable, the added complexity is not worth the benefits.

**Comment by perogi** [comment]
@rspilker - Hi Roel, I noticed that you wrote 'I did have a look at the code, and although doable, the added complexity is not worth the benefits".  I cloned the repo and am trying to figure out where to add code to implement that example you wrote in https://github.com/rzwitserloot/lombok/issues/1999#issuecomment-456216505 - specifically ` if (values != null) {`

For my team's use case, ignoring the parameter `values` if it is null is an acceptable implementation and fits what we need.

Any hints on where I should look? I updated a the test cases for our solution but really struggling on where in the code to look/change.

Thank you very much.
perogi.
